modules/vpc/variables.tf

```
@@ -1,85 +1,152 @@
-variable "networks" {
+variable "subnetworks" {
   description = <<-EOF
-  Map of networks, a minimal example:
+  A map containing subnetworks configuration. Subnets can belong to different regions.
+  List of available attributes of each subnetwork entry:
+  - `subnetwork_name` : Name of the subnetwork.
+  - `create_subnetwork` : Boolean value to control the creation or reading of the subnetwork. If set to `true` - this will create the subnetwork. If set to `false` - this will read a subnet with provided information.
+  - `ip_cidr_range` : A string that contains the subnetwork to create. Only IPv4 format is supported.
+  - `region` : Region where to configure or import the subnet.
 
+  Example:
   ```
-  {
-    "my-vpc" = {
-      name            = "my-vpc"
-      subnetwork_name = "my-subnet"
-      ip_cidr_range   = "192.168.1.0/24"
-    }
-  }
-  ```
-
-  An advanced example:
-
-  ```
-  {
-    "my-vpc" = {
-      name            = "my-vpc"
-      subnetwork_name = "my-subnet"
-      ip_cidr_range   = "192.168.1.0/24"
-      allowed_sources = ["209.85.152.0/22"]
-      log_metadata    = "INCLUDE_ALL_METADATA"
-      mtu             = 1500
-      routing_mode    = "REGIONAL"
+  subnetworks = {
+    my-sub = {
+      subnetwork_name = "my-sub"
+      create_subnetwork = true
+      ip_cidr_range = "192.168.0.0/24"
+      region = "us-east1"
     }
   }
   ```
+  EOF
+  default     = {}
+  type = map(object({
+    subnetwork_name   = string
+    create_subnetwork = bool
```

```suggestion
    create_subnetwork = optional(bool, true)
```
I think we can stick to the same default as for the network.